dock/run.sh

```diff
@@ -34,6 +34,9 @@ for f in $folders ; do test -d $f && FOLDERS+=" -v $f:$f"; done
 FOLDERS+=" -v `pwd`/build/dock-builD:`pwd`/build"
 FOLDERS+=" -v `pwd`/build/dock-node_modules:`pwd`/node_modules"
 mkdir -p build/dock-builD
+mkdir -p build/dock-node_modules
```

Comment:
what are these for? 
In any case, you have to do `./dock/run.sh npm install` to get all required modules. This copies only few. We _could_ copy a lot of modules (just to save time for npm install) - but must skip all those with native code.

Reply:
We only need them for openzeppelin & @0x deps we have in the contracts currently

Reply:
no. if you want to do test or whatever inside the dock, you need other modules too. don't start copy partial modules around..
we must do `./dock/run.sh npm install`.
(the `mkdir` you added above IS required, but the `cp` below are not)

Reply:
`./dock/run.sh npm install` doesn't work. It probably needs to be run as root but I'm not going to run it with sudo.

Reply:
it does work, but only if the dock-node_modules folder exists before (the docker uses existing folder, but if it doesn't exists, it creates one as *root*, which breaks the scripts.
that's why I said the mkdir is required, but the rest isn't.
I just did such "docker/run npm install" on a clean environment .

Reply:
Now it works, but because of the mappings of the directories in we give the docker, it installs the modules in `build/dock-node_modules` but updates `package.json` and `package-lock.json` outside of docker which is a big issue.

Reply:
done